src/kaleidoscope/plugin/IdleLEDs.cpp

```diff
@@ -25,23 +26,33 @@ uint16_t IdleLEDs::idle_time_limit = 600; // 10 minutes
 uint32_t IdleLEDs::end_time_;
 
 EventHandlerResult IdleLEDs::beforeEachCycle() {
-  if (!::LEDControl.paused &&
-      (Kaleidoscope.millisAtCycleStart() >= end_time_)) {
+  if (Kaleidoscope.millisAtCycleStart() < end_time_)
+    return EventHandlerResult::OK;
+
+  if (::Kaleidoscope.has_leds && !::LEDControl.paused) {
```

Comment:
I think this is a small change that can land without the underglow PR

Reply:
I started to extract this into a separate PR, but I decided not to for the time being. Reading back the discussion here, I have the feeling it will not be needed. The change is here to support `UnderglowControl`, but if we bake underglow control into `LEDControl`, then this is a needless exercise, at least in its current form.